command/runner.go

```diff
@@ -203,3 +219,69 @@ func (r *cmdWaitReadCloser) Close() error {
 	r.closed = true
 	return nil
 }
+
+// Close the pipe and wait until the consumer program exits
+func (p pagerPipe) Close() error {
+	err := p.writer.Close()
+	<-p.cc
+	return err
+}
+
+func (p pagerPipe) Writer() io.Writer {
+	return p.writer
+}
+
+func defaultPager() string {
+	if p := os.Getenv("PAGER"); p != "" {
+		return p
+	}
+	if _, err := exec.LookPath("less"); err == nil {
+		return "less -R"
```

Comment:
I think we should default to using `-F` flag too.


```suggestion
		return "less -RF"
```

Reply:
Hmm, isn't `paging: always` a little confusing if the default is to use `less -RF`?

Reply:
I think `paging: auto` should really only mean that kubecolor is doing the screen size detection, not the pager.

I think it's okay to have "always" mean "pipe through a pager regardless of screen size, unless it's really a bad idea to pipe to a paper" (like if stdout is not the terminal). Isn't that what other tools mean by "always"?

One thing we can do is use "on" / "off" for now and make these aliases for "always" / "never" if we ever introduce an "auto" option.

Reply:
@applejag what do you think of my suggestion here? I think this is the only thing that still needs to be decided before we can finish this PR.

Reply:
I'm being more and more reluctant to using "always", as kubecolor uses a whitelist of commands that support the pager. The "always" value does not bypass this whitelist.

I'd be nice to avoid using a boolean to keep it an open-ended enum. On the other hand, using "on"/"off" feels so disconnected.

> I think `paging: auto` should really only mean that kubecolor is doing the screen size detection, not the pager.

That just feels like unnecessary extra buffering and logic that needs to added to kubecolor. Logic that the pager already takes care of.

Except for that, `auto` would still mean:

- when using a white-listed command
- when stdout is a tty
- if `pager` config is unset, then try use `less` or `more`

It doesn't *also* need to do it's own terminal hight check.

Is there some scope that `always` would practically fit? Should `auto` take the current `always`'s place, and then redefine `always` to mean "ignore the command white list"? Or maybe "run even when no tty"?

Maybe that decision could come later, and we just settle now for only having `auto` and `never`

Reply:
> Is there some scope that always would practically fit? Should auto take the current always's place, and then redefine always to mean "ignore the command white list"? Or maybe "run even when no tty"?

I think you should just run `kubecolor | pager` if you want to ignore the command whitelist. The reason I made this PR was precisely to have detection of tty and subcommand.

> That just feels like unnecessary extra buffering and logic that needs to added to kubecolor. Logic that the pager already takes care of.

Agreed.

So I don't think we will need more than two options. But I'll call it `auto` and `never` for now.

Reply:
Alright sticking with `auto` and `never`